### DISCUSSION TITLE
nspanel_xx_code /*.txt

### DISCUSSION THREAD

**Discussion by fordcrews** [Blackymas/NSPanel_HA_Blueprint#1029] [General]
Where do these come from?    

Are they created in the nextion editor and compiled into the .tft file?

I saw the program.s in there, but couldn't find these find the rest but they seem to follow the pages in the editor.

I was looking at the yaml files and am just trying to work out how it's all related.

**Comment by edwardtfn** [comment]
Those files are created by https://github.com/MMMZZZZ/Nextion2Text and are basically a plain text export of the HMI. Not all the information is there, but it is quite useful to visualize the changes in GitHub, as we cannot see the changes on the binaries (.HMI or .tft).

**Reply by fordcrews** [reply]
Cool I kept looking for an export button/save as text/etc and thought I must be over looking it.
